include/mata/nfa.hh

```diff
@@ -461,7 +645,30 @@ public:
 
         return transitionrelation[state];
     } // operator[] }}}
-};
+
+private:
+    using StateBoolArray = std::vector<bool>; ///< Bool array for states in the automaton.
+
+    /**
+     * Compute reachability of states.
+     * @return Bool array for reachable states (from initial states): true for reachable, false for unreachable states.
+     */
+    StateBoolArray compute_reachability() const;
+
+    /**
+     * Add transitions to the trimmed automaton.
+     * @param original_to_new_states_map Map of old states to new trimmed automaton states.
+     * @param trimmed_aut The new trimmed automaton.
+     */
+    void add_trimmed_transitions(const StateMap<State>& original_to_new_states_map, Nfa& trimmed_aut);
+
+    /**
+     * Get new trimmed automaton.
+     * @param original_to_new_states_map Map of old states to new trimmed automaton states.
+     * @return Newly initialized trimmed automaton.
+     */
+    Nfa initialize_trimmed_aut(const StateMap<State>& original_to_new_states_map);
```

Comment:
Weird naming, I would think that "initialize" will no return anything at all. Can we name this to something more intuitive? Maybe simply `create_trimmed_automaton`?

Reply:
I can see what you mean. `create_trimmed_automaton` seems like a good replacement which better describes what the function does.